src/pack.c

```diff
@@ -1164,28 +1211,42 @@ int git_packfile_alloc(struct git_pack_file **pack_out, const char *path)
  *
  ***********************************************************/
 
-static off64_t nth_packed_object_offset(const struct git_pack_file *p, uint32_t n)
+static off64_t nth_packed_object_offset(struct git_pack_file *p, uint32_t n)
 {
-	const unsigned char *index = p->index_map.data;
-	const unsigned char *end = index + p->index_map.len;
+	const unsigned char *index, *end;
+	int error;
+	uint32_t off32;
+	off64_t off;
+
+	if ((error = git_mutex_lock(&p->lock)) < 0)
+		return error;
```

Comment:
`git_mutex_*` is one of the outlier set of functions that are prefixed `git_` but don't set error messages when they fail.  They rely on the caller to know that they need to do that and use `GIT_ERROR_OS`.  This stinks and we should fix this.  It should be `p_mutex_lock` or something.  In the meantime, would you mind:

```suggestion
	if ((error = git_mutex_lock(&p->lock)) < 0) {
		git_error_set(GIT_ERROR_OS, "failed to lock packfile reader");
		return error;
	}
```

Reply:
noted! will do that in the future.